allennlp/modules/token_embedders/pretrained_transformer_embedder.py

```diff
@@ -122,6 +125,18 @@ def __init__(
         self._num_added_tokens = self._num_added_start_tokens + self._num_added_end_tokens
 
         if not train_parameters:
+            # Calling transformer_model.eval() won't change anything now,
+            # so we have to explicitly set training = False
+            self.transformer_model.training = False
+
+            # Override train in transformer_model to prevent it from changing modes
+            def _train(self, mode):
+                return self
+
+            setattr(
+                self.transformer_model, "train", types.MethodType(_train, self.transformer_model)
+            )
```

Comment:
This makes me a bit nervous. Another option would be to add something like this to the `.forward()` method below (after line 182):

```python
if self.train_parameters and self.training and self.transformer_model.training:
    self.transformer_model.eval()
```

Reply:
I like this approach much better, thanks for the suggestion :)

Reply:
One issue with this though is that inspecting `.training` will not alway produce the expected result.